### PR TITLE
Add Syntax API Code Gen related functionality into the extension

### PR DESCRIPTION
**Pull request by kdsuneraavinash**
## Purpose
$subject

Peek 2021-04-20 16-29

## Goals
Add foundation files and a sample use case of integrating Syntax API code gen feature.
- A new command to copy the java code required to generate the current file to clipboard

## Approach
Add `vs-syntax-api-calls-gen` module and register its commands
Add syntax API code gen related functionality to language client
Add some tests

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -66,6 +67,11 @@
                 "command": "ballerinaCompilerToolkit.visualizeSyntaxTree",
                 "title": "Syntax Tree Visualizer",
                 "category": "Ballerina Compiler Toolkit"
+            },
+            {
+                "command": "ballerinaCompilerToolkit.copySyntaxApiCalls",
+                "title": "Copy Syntax-API Calls to Generate this Code",
```
Comment (nadeeshaan) [inline package.json:73]: Shall we use `Copy Syntax API calls to Clipboard` instead?

**Review by nadeeshaan — CHANGES_REQUESTED** [review]

**Review by SumayyaZiyad — APPROVED** [review]
LGTM :)

File: src/core/extended-language-client.ts [hunk 2]
```diff
@@ -18,8 +18,8 @@
  *
  */
 
-import { Selection, Uri } from "vscode";
-import { LanguageClient } from "vscode-languageclient";
+import {Selection, Uri} from "vscode";
+import {LanguageClient} from "vscode-languageclient";
```
Comment (SumayyaZiyad) [inline src/core/extended-language-client.ts:22]: The space between the curly braces and the imports are a convention we followed throughout the code base. Shall we stick to it?

**Comment by kdsuneraavinash** [comment]
The CI seems to be pulling slp4 but since syntax API gen is not available in slp4 the tests are failing

**Comment by SumayyaZiyad** [comment]
> The CI seems to be pulling slp4 but since syntax API gen is not available in slp4 the tests are failing

Had the same issue with the visualizer till slp4 was available. Based on the current workflow, until slp5 is available, I guess we will have to drop the tests on the main branch. 
At the same time, will also look into using a build from the master  instead.

**Comment by kdsuneraavinash** [comment]
@SumayyaZiyad yeah, it should work with slp5
I disabled tests cases for now and opened #5

**Comment by SumayyaZiyad** [comment]
> @SumayyaZiyad yeah, it should work with slp5
> I disabled tests cases for now and opened #5

The preview is available now right. So I can update the GH Actions to pull slp5 instead. If so we don't have to disable the test cases.
But we still have to manually keep updating the package being pulled - will try to figure out an alternative anyways. For now, will pull slp5. Sending a PR with the updated actions now.
Sorry about the mess :)

**Comment by SumayyaZiyad** [comment]
> @SumayyaZiyad yeah, it should work with slp5
> I disabled tests cases for now and opened #5

Noted. Will re-enable the tests, once the approach to pulling the pack is figured out!

**Review by nadeeshaan — APPROVED** [review]
LGTM

**Pull request merged**